pkg/machinery/marker.go

```diff
@@ -38,7 +40,7 @@ type Marker struct {
 }
 
 // NewMarkerFor creates a new marker customized for the specific file
-// Supported file extensions: .go, .yaml, .yml
+// Supported file extensions: .go, .mod, .work, .yaml, .yml
```

Comment:
why do we need the add the .work extension ?

Reply:
The .work extension is necessary because of the scaffolding in the `go.work` file that adds api modules as they are generated:

```go.work
go 1.18

use ./api/v1alpha1 // <<=== this is generated for each kubebuilder create api request

//+kubebuilder:scaffold:apis // this is the scaffolding we need to read

use .
```

Reply:
On that note, I noticed the Dockerfile also needs to be optimized with workspaces, so I started to introduce a Dockerfile Marker Support as well